Framework/API/src/ExperimentInfo.cpp

```diff
@@ -197,7 +201,31 @@ makeDetectorInfo(const Instrument &oldInstr, const Instrument &newInstr) {
     for (size_t i = 0; i < numDets; ++i)
       if (newInstr.isMonitorViaIndex(i))
         monitors.push_back(i);
-    return Kernel::make_unique<Beamline::DetectorInfo>(numDets, monitors);
+    std::vector<Eigen::Vector3d> positions;
+    std::vector<Eigen::Quaterniond> rotations;
+    const auto &detIDs = newInstr.getDetectorIDs();
+    for (const auto &id : detIDs) {
+      const auto &det = newInstr.getDetector(id);
+      // In the case of RectangularDetectorPixel the position is also affected
+      // by the parameters scalex and scaly, but `getPos()` takes that into
+      // account (if no DetectorInfo is set in the ParameterMap).
+      positions.emplace_back(toVector3d(det->getPos()));
+      rotations.emplace_back(toQuaterniond(det->getRotation()));
+      pmap->clearParametersByName(ParameterMap::pos(), det.get());
```

Comment:
This is good. I wonder if we could make this `makeDetectorInfo` method more concise by shifting some of these `clearParametersByName` calls into a single `ParameterMap::clearLegacyGeometryParameters` (or something like that) with the added benefit that you could provide something like a `const IDetector&` as the argument, and avoid all these `shared_ptr<T>::get()` calls?